reconcile/utils/unleash.py

```diff
@@ -1,117 +1,92 @@
-    return True
+class CacheDict(BaseCache):
+    def __init__(self):
+        self.cache = {}
 
+    def set(self, key: str, value: Any):
+        self.cache[key] = value
 
-@defer
-def get_feature_toggle_state(integration_name, defer=None):
-    api_url = os.environ.get("UNLEASH_API_URL")
-    client_access_token = os.environ.get("UNLEASH_CLIENT_ACCESS_TOKEN")
-    if not (api_url and client_access_token):
-        return True
+    def mset(self, data: dict):
+        self.cache.update(data)
 
-    # create temporary cache dir
-    cache_dir = tempfile.mkdtemp()
-    defer(lambda: shutil.rmtree(cache_dir))
+    def get(self, key: str, default: Optional[Any] = None):
+        return self.cache.get(key, default)
 
-    # hide INFO logging from UnleashClient
-    with log_lock:
-        with toggle_logger():
-            # create Unleash client
-            headers = {"Authorization": f"Bearer {client_access_token}"}
-            client = UnleashClient(
-                url=api_url,
-                app_name="qontract-reconcile",
-                custom_headers=headers,
-                cache_directory=cache_dir,
-            )
-            client.initialize_client()
+    def exists(self, key: str):
+        return key in self.cache
 
-            # get feature toggle state
-            state = client.is_enabled(
-                integration_name, fallback_function=get_feature_toggle_default
-            )
-            client.destroy()
-
-        return state
-
-
-@defer
-def get_feature_toggles(api_url, client_access_token, defer=None):
-    # hide INFO logging from UnleashClient
-    with toggle_logger():
-        # create temporary cache dir
-        cache_dir = tempfile.mkdtemp()
-        defer(lambda: shutil.rmtree(cache_dir))
-
-        # create Unleash client
-        headers = {"Authorization": f"Bearer {client_access_token}"}
-        client = UnleashClient(
-            url=api_url,
-            app_name="qontract-reconcile",
-            custom_headers=headers,
-            cache_directory=cache_dir,
-        )
-        client.initialize_client()
-        defer(client.destroy)
-
-    return {
-        k: "enabled" if v.enabled else "disabled" for k, v in client.features.items()
-    }
-
-
-@defer
-def get_unleash_strategies(api_url, token, strategy_names, defer=None):
-    # create strategy mapping
-    unleash_strategies = {name: strategies.Strategy for name in strategy_names}
-
-    # create temporary cache dir
-    cache_dir = tempfile.mkdtemp()
-    defer(lambda: shutil.rmtree(cache_dir))
-
-    # hide INFO logging from UnleashClient
-    with log_lock:
-        with toggle_logger():
-            # create Unleash client
-            headers = {"Authorization": f"Bearer {token}"}
+    def destroy(self):
+        self.cache = {}
+
+
+class DisableClusterStrategy(Strategy):
+    def load_provisioning(self) -> list:
+        return [x.strip() for x in self.parameters["cluster_name"].split(",")]
```

Comment:
tiny nit: since `cluster_name` can hold a list of clusters we could name it `clusters_names` or `cluster_names_csv`

Reply:
The unleash UI abstracts the list, however, that name is directly displayed there. You enter one cluster name at a time. If I change it to be plular, it might not be easy to understand in Unleash, since the name would suggest you can enter multiple names.
So, I'd prefer to leave it as is...

Reply:
thanks for the clarification!